src/appengine.h

```diff
@@ -19,6 +19,7 @@ class AppEngine {
   virtual bool run(const App& app) = 0;
   virtual void remove(const App& app) = 0;
   virtual bool isRunning(const App& app) const = 0;
+  virtual std::string containerDetails() const = 0;
```

Comment:
maybe `getRunningApps()` instead of `containerDetails`. Also, returning Json can be more flexible, clients can construct a string/output they need in their specific use-case.

Reply:
sure, just named it using the original function it replaced....will do runningApps()

Reply:
sure I'll return Json (the original code was returning a string)

Reply:
what format does the json value needs?